servlet-feature-pack/src/main/resources/modules/system/layers/base/org/wildfly/clustering/web/container/main/module.xml

```diff
@@ -20,11 +21,28 @@
   ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
   ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
   -->
-<jboss-web xmlns="http://www.jboss.com/xml/ns/javaee"
-           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-           xsi:schemaLocation="http://www.jboss.com/xml/ns/javaee http://www.jboss.org/j2ee/schema/jboss-web_12_0.xsd"
-           version="12.0">
-    <replication-config>
-        <cache-name>web.database-persistence</cache-name>
-    </replication-config>
-</jboss-web>
+
+<module xmlns="urn:jboss:module:1.8" name="org.wildfly.clustering.web.container">
+    <properties>
+        <property name="jboss.api" value="private"/>
+    </properties>
+
+    <resources>
+        <artifact name="${org.wildfly:wildfly-clustering-web-container}"/>
+    </resources>
+
+    <dependencies>
+        <module name="javax.api"/>
+        <module name="org.jboss.as.clustering.common"/>
+        <module name="org.jboss.as.controller"/>
+        <module name="org.jboss.as.server"/>
+        <module name="org.jboss.dmr"/>
+        <module name="org.jboss.logging"/>
+        <module name="org.jboss.modules"/>
+        <module name="org.jboss.msc"/>
+        <module name="org.wildfly.clustering.service"/>
+        <module name="org.wildfly.clustering.web.undertow" services="import" optional="true"/>
+        <module name="org.wildfly.common"/>
+        <module name="org.wildfly.security.elytron-private"/>
+    </dependencies>
```

Comment:
The following modules are not used to cover a direct dependency of this module:
- org.jboss.as.controller
- org.wildfly.security.elytron-private
- org.jboss.dmr
- org.wildfly.common
- org.jboss.as.server
- org.jboss.logging
- org.wildfly.clustering.service
- org.wildfly.clustering.web.undertow

I checked only direct dependencies, I have not checked if they are used to supply a service loaded by this module or if they are required due to an external activation. That requires a wide knowledge of this subsystem, but maybe the list of required dependencies can be reduced using the above list.

Reply:
Thanks for the analysis - some of these are certainly required, but I'll verify the others.